firebase-perf/src/main/java/com/google/firebase/perf/util/Timer.java

```diff
@@ -16,68 +16,75 @@
 
 import android.os.Parcel;
 import android.os.Parcelable;
+import android.os.SystemClock;
 import androidx.annotation.NonNull;
 import com.google.android.gms.common.util.VisibleForTesting;
 import java.util.concurrent.TimeUnit;
 
-/** A Timer class provides both wall-clock (epoch) time and high resolution time (nano time). */
+/** A Timer class provides both wall-clock (epoch) time and monotonic time (elapsedRealtime). */
 public class Timer implements Parcelable {
 
-  /** Wall-clock time or epoch time in microseconds, */
-  private long timeInMicros;
   /**
-   * High resolution time in nanoseconds. High resolution time should only be used to calculate
-   * duration or latency. It is not wall-clock time.
+   * Wall-clock time or epoch time in microseconds. Do NOT use for duration because wall-clock is
+   * not guaranteed to be monotonic: it can be set by the user or the phone network, thus it may
+   * jump forwards or backwards unpredictably. {@see SystemClock}
    */
-  private long highResTime;
+  private long wallClockMicros;
+  /**
+   * Monotonic time measured in the {@link SystemClock#elapsedRealtime()} timebase. Only used to
+   * compute duration between 2 timestamps in the same timebase. It is NOT wall-clock time.
+   */
+  private long elapsedRealtimeNanos;
 
   /**
    * Construct Timer object using System clock. Make it package visible to be only accessible from
    * com.google.firebase.perf.util.Clock.
    */
   public Timer() {
-    timeInMicros = TimeUnit.MILLISECONDS.toMicros(System.currentTimeMillis());
-    highResTime = System.nanoTime();
+    wallClockMicros = TimeUnit.MILLISECONDS.toMicros(System.currentTimeMillis());
+    elapsedRealtimeNanos = SystemClock.elapsedRealtimeNanos();
   }
 
   /**
-   * Construct a Timer object with input wall-clock time, assume high resolution time is same as
-   * wall-clock time.
+   * Returns a new Timer object with input elapsedRealtime. Uses current wall-clock as a reference
+   * to extrapolate the wall-clock at the time of the input elapsedRealtime.
    *
-   * @param time wall-clock time in microseconds
+   * @param elapsedRealtimeNanos timestamp in the {@link SystemClock#elapsedRealtime()} timebase
    */
-  @VisibleForTesting
-  public Timer(long time) {
-    this.timeInMicros = time;
-    highResTime = TimeUnit.MICROSECONDS.toNanos(time);
+  public static Timer of(long elapsedRealtimeNanos) {
```

Comment:
Moved to above the constructors, because I want to eventually make all constructors private and only allow new instance creation through `public static` factory methods, due to Effective Java [Item 1: Consider static factory methods instead of constructors](http://go/ej3e-1)